packages/@angular/cli/models/architect-command.ts

```diff
@@ -229,4 +230,35 @@ export abstract class ArchitectCommand<T = any> extends Command<T> {
       tap((architect: Architect) => this._architect = architect),
     );
   }
+
+  private _makeTargetSpecifier(options: ArchitectCommandOptions): TargetSpecifier {
+    let project, target, configuration, overrides;
+
+    if (options.target) {
+      [project, target, configuration] = options.target.split(':');
+
+      overrides = { ...options };
+      delete overrides.target;
+    } else {
+      project = options.project;
+      target = this.target;
+      configuration = options.configuration;
+      if (!configuration && options.prod) {
```

Comment:
I started that way, but then it turned out that not having prod is specific to only one command ou of seven. 

Having to massage the options in the six individual commands instead of doing it in `ArchitectCommand` seems like the better approach, especially since that sort of processing also needs to happen within `ArchitectCommand` for initialization and validation, which happens before running the command proper.

Reply:
It's awkward that it works on commands it should not but we can correct that moving forward.